src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const expensesService = require('../services/expenses.services');
+const userService = require('../services/users.services');
+const { expensesFilter } = require('../utils/expensesFilter');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    categories,
+  } = req.query;
+
+  const expenses = expensesFilter(userId, from, to, categories);
+
+  res.send(expenses);
+};
+
+const getById = (req, res) => {
+  const { id } = req.params;
+  const searcedExpense = expensesService.getById(+id);
+
+  if (!id) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!searcedExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.status(200).send(searcedExpense);
+};
+
+const post = (req, res) => {
+  const { userId, ...rest } = req.body;
+
+  if (!userService.getById(userId)) {
+    return res.sendStatus(400);
+  }
+
+  const newExpense = {
+    id: Date.now(),
+    userId,
+    ...rest,
+  };
+
+  expensesService.add(newExpense);
+
+  res.status(201).send(newExpense);
+};
+
+const update = (req, res) => {
+  const { id } = req.params;
+  const expense = expensesService.getById(+id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const updateExpense = expensesService.update(expense, req.body);
```

Comment:
What if I send you wrong data to update the expense, it would break I suppose. You need to check req.body before implementing it